src/main/java/com/azure/cosmos/examples/diagnostics/async/CosmosDiagnosticsQuickStartAsync.java

```diff
@@ -0,0 +1,284 @@
+import com.azure.cosmos.models.CosmosDatabaseResponse;
+import com.azure.cosmos.models.CosmosItemRequestOptions;
+import com.azure.cosmos.models.CosmosItemResponse;
+import com.azure.cosmos.models.CosmosQueryRequestOptions;
+import com.azure.cosmos.models.PartitionKey;
+import com.azure.cosmos.models.ThroughputProperties;
+import com.azure.cosmos.util.CosmosPagedFlux;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.UUID;
+
+public class CosmosDiagnosticsQuickStartAsync {
+
+    private CosmosAsyncClient client;
+
+    private final String databaseName = "AzureSampleFamilyDB";
+    private final String containerName = "FamilyContainer";
+    private final String documentId = UUID.randomUUID().toString();
+    private final String documentLastName = "Witherspoon";
+
+    private CosmosAsyncDatabase database;
+    private CosmosAsyncContainer container;
+
+    private final static Logger logger = LoggerFactory.getLogger(CosmosDiagnosticsQuickStartAsync.class);
+
+    public void close() {
+        client.close();
+    }
+
+    public static void main(String[] args) {
+        CosmosDiagnosticsQuickStartAsync quickStart = new CosmosDiagnosticsQuickStartAsync();
+
+        try {
+            logger.info("Starting ASYNC main");
+            quickStart.diagnosticsDemo();
+            logger.info("Demo complete, please hold while resources are released");
+        } catch (Exception e) {
+            logger.error("Cosmos getStarted failed with", e);
+        } finally {
+            logger.info("Shutting down");
+            quickStart.shutdown();
+        }
+    }
+
+    private void diagnosticsDemo() throws Exception {
+
+        logger.info("Using Azure Cosmos DB endpoint: " + AccountSettings.HOST);
+
+        //  Create sync client
+        client = new CosmosClientBuilder()
+            .endpoint(AccountSettings.HOST)
+            .key(AccountSettings.MASTER_KEY)
+            .consistencyLevel(ConsistencyLevel.EVENTUAL)
+            .contentResponseOnWriteEnabled(true)
+            .buildAsyncClient();
+
+
+        createDatabaseIfNotExists();
+        createContainerIfNotExists();
+
+        createDocument();
+        readDocumentById();
+        queryDocuments();
+        replaceDocument();
+        upsertDocument();
+    }
+
+    // Database Diagnostics
+    private void createDatabaseIfNotExists() throws Exception {
+        logger.info("Creating database " + databaseName + " if not exists...");
+
+        //  Create database if not exists
+        Mono<CosmosDatabaseResponse> databaseResponseMono = client.createDatabaseIfNotExists(databaseName);
+        CosmosDatabaseResponse cosmosDatabaseResponse = databaseResponseMono.flatMap(databaseResponse -> {
+            CosmosDiagnostics diagnostics = databaseResponse.getDiagnostics();
+            logger.info("Create database diagnostics : {}", diagnostics);
```

Comment:
if we are doing blocking call at the end, I would capture the request diagnostics outside the flatMap, it is both simpler and also if customer is using non-async-logger this won't happen on the netty io thread.

Reply:
I see, the reason I added them to the `flatMap` call is because that is the natural way of using async API. The only reason we have blocking calls is the make the samples easier and quicker. 
I can use `doOnNext()` if that would not cause the netty IO thread to come into the picture. 

But if are going to get diagnostics after block(), then I don't think we should even have an async sample, because from the sample point of view, it looks exactly the same. 
Thoughts ?

Reply:
I see your point. my worry is those end users who will just copy and paste code. both flatMap and doOnNext will do the same with response to the threading.

IMO, The advance users seeing the code can figure that out how to do chaining. our samples should target the naive users.

Reply:
Yeah, that is true though. If we are targeting naive users, then it makes sense to change it, will do. Thanks.